### ISSUE TITLE
 skimage FTBFS on multiple architectures testsuite failures.

### ISSUE THREAD

**Issue by tillea** [scikit-image/scikit-image#2398]
## Description
When building skimage on Debian autobuilders the test suite fails on several architectures.
In the [bug log](https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=840589#35) it is explained that when checking with gdb it is just by chance that the failure does not happen with amd64.  The problematic commit is d29831882daef9b7d7691becc3669349d642b547.  The variable firstbg is only initialized conditionally in the first loop. It depends on the array values and the and background value. Still it is used as the starting pointer on the second loop, even if uninitialized. At best the pointer point to the wrong address, at worst it generates an out of bound access causing the segmentation fault.

## Way to reproduce
See [bug log](https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=840589#35)

- [ ] Debian Linux 
- [ ] 0.12.3

**Comment by yarikoptic** [comment]
tentative fix in @2399

**Comment by stefanv** [comment]
@tillea Thanks for reporting.

@yarikoptic In the above comment, you refer to `@2399` --- is this a commit, and where can I find it?

**Comment by soupault** [comment]
@stefanv https://github.com/scikit-image/scikit-image/pull/2399

**Comment by yarikoptic** [comment]
eh -- #2399 fixed that issue but it seems that new ones (either as a result of my fix or not) lead to new failures:
```

======================================================================
FAIL: skimage.feature.tests.test_orb.test_descriptor_orb
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/lib/python2.7/dist-packages/nose/case.py", line 197, in runTest
    self.test(*self.arg)
  File "/«PKGBUILDDIR»/debian/tmp/usr/lib/python2.7/dist-packages/skimage/feature/tests/test_orb.py", line 99, in test_descriptor_orb
    detector_extractor.descriptors[100:120, 10:20])
  File "/usr/lib/python2.7/dist-packages/numpy/testing/utils.py", line 322, in assert_equal
    return assert_array_equal(actual, desired, err_msg, verbose)
  File "/usr/lib/python2.7/dist-packages/numpy/testing/utils.py", line 813, in assert_array_equal
    verbose=verbose, header='Arrays are not equal')
  File "/usr/lib/python2.7/dist-packages/numpy/testing/utils.py", line 739, in assert_array_compare
    raise AssertionError(msg)
AssertionError: 
Arrays are not equal

(mismatch 46.5%)
 x: array([[False,  True,  True,  True, False,  True, False,  True, False,
         True],
       [ True,  True,  True, False, False,  True, False, False,  True,...
 y: array([[ True, False, False, False, False, False, False, False,  True,
         True],
       [False,  True,  True,  True, False,  True, False,  True, False,...

======================================================================
FAIL: skimage.filters.rank.tests.test_rank.test_all
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/lib/python2.7/dist-packages/nose/case.py", line 197, in runTest
    self.test(*self.arg)
  File "/«PKGBUILDDIR»/debian/tmp/usr/lib/python2.7/dist-packages/skimage/filters/rank/tests/test_rank.py", line 16, in test_all
    check_all()
  File "/«PKGBUILDDIR»/debian/tmp/usr/lib/python2.7/dist-packages/skimage/_shared/testing.py", line 232, in inner
    result = func(*args, **kwargs)
  File "/«PKGBUILDDIR»/debian/tmp/usr/lib/python2.7/dist-packages/skimage/filters/rank/tests/test_rank.py", line 83, in check_all
    rank.entropy(image, selem))
  File "/usr/lib/python2.7/dist-packages/numpy/testing/utils.py", line 322, in assert_equal
    return assert_array_equal(actual, desired, err_msg, verbose)
  File "/usr/lib/python2.7/dist-packages/numpy/testing/utils.py", line 813, in assert_array_equal
    verbose=verbose, header='Arrays are not equal')
  File "/usr/lib/python2.7/dist-packages/numpy/testing/utils.py", line 739, in assert_array_compare
    raise AssertionError(msg)
AssertionError: 
Arrays are not equal

(mismatch 1.28%)
 x: array([[ 1.584963,  2.      ,  2.      ,  2.      ,  2.      ,  2.      ,
         2.      ,  2.      ,  2.      ,  2.      ,  2.      ,  2.      ,
         2.      ,  2.      ,  2.      ,  2.      ,  2.      ,  2.      ,...
 y: array([[ 1.584963,  2.      ,  2.      ,  2.      ,  2.      ,  2.      ,
         2.      ,  2.      ,  2.      ,  2.      ,  2.      ,  2.      ,
         2.      ,  2.      ,  2.      ,  2.      ,  2.      ,  2.      ,...
```

I will try to pin point them...

**Comment by yarikoptic** [comment]
actually -- would be nice if someone from @scikit-image/core could have a look at those -- since seems to rely on assumptions of algorithms operation (hardcoded target value) which I am not familiar with

**Comment by soupault** [comment]
Fixed via https://github.com/scikit-image/scikit-image/pull/2399.

**Comment by soupault** [comment]
Fixed via #2399.
